syntax = "proto3";

package cafm.identity.auth;

message AuthAppService_authenticateUserByNameAndPasswordRequest {
  string name = 1;
  string password = 2;
}

message AuthAppService_authenticateUserByNameAndPasswordResponse {
  string token = 1;
}

message AuthAppService_isAuthenticatedRequest {
  string token = 1;
}

message AuthAppService_isAuthenticatedResponse {
  bool response = 1;
}

message AuthAppService_logoutRequest {
  string token = 1;
}

message AuthAppService_logoutResponse {}

service AuthAppService {
  rpc authenticateUserByNameAndPassword (AuthAppService_authenticateUserByNameAndPasswordRequest) returns (AuthAppService_authenticateUserByNameAndPasswordResponse) {}
  rpc isAuthenticated (AuthAppService_isAuthenticatedRequest) returns (AuthAppService_isAuthenticatedResponse) {}
  rpc logout (AuthAppService_logoutRequest) returns (AuthAppService_logoutResponse) {}
}